File: reporters/golang/tt/reporter/crypto.go
package reporter

import (
	"crypto/aes"
	"crypto/cipher"
	"crypto/rand"
	"crypto/sha256"
	"encoding/base64"
	"io"
)

// Encrypt encrypts plaintext using AES-GCM with the supplied password.
func encrypt(plaintext string, password string) (string, error) {
    // Encode password as UTF-8
    pwUtf8 := []byte(password)
    // Hash the password
    pwHash := sha256.Sum256(pwUtf8)
    // Get 96-bit random iv
    iv := make([]byte, 12)
    if _, err := io.ReadFull(rand.Reader, iv); err != nil {
        return "", err
    }
    // Specify algorithm to use
    block, err := aes.NewCipher(pwHash[:])
    if err != nil {
        return "", err
    }
    // Create cipher instance
    gcm, err := cipher.NewGCM(block)
    if err != nil {
        return "", err
    }
    // Encrypt plaintext
    ciphertext := gcm.Seal(nil, iv, []byte(plaintext), nil)
    // Encode iv+ciphertext as base64
    encrypted := append(iv, ciphertext...)
    return base64.StdEncoding.EncodeToString(encrypted), nil
}

File: reporters/python/tracethat/crypto.py
import base64
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from hashlib import sha256
import os

def encrypt(plaintext, password):
    # Encode password as bytes
    pw_bytes = password.encode('utf-8')
    
    # Hash the password using SHA-256
    pw_hash = sha256(pw_bytes).digest()
    
    # Generate a random 12-byte IV
    iv = os.urandom(12)
    
    # Create the AES-GCM cipher
    cipher = Cipher(algorithms.AES(pw_hash), modes.GCM(iv))
    encryptor = cipher.encryptor()
    
    # Encrypt the plaintext
    ciphertext = encryptor.update(plaintext.encode('utf-8')) + encryptor.finalize()
    
    # Concatenate the IV and ciphertext, then encode as base64
    encrypted = base64.b64encode(iv + ciphertext + encryptor.tag).decode('utf-8')
    
    return encrypted